rmw_iceoryx_cpp/src/internal/iceoryx_name_conversion.cpp

```diff
@@ -49,13 +49,15 @@ std::tuple<std::string, std::string, std::string> get_service_description_elemen
   }
 
   // Could check more detailed!!
-  auto service = type_name.substr(0, pos_msg);
-  auto pos_type_name = topic_name.find(type_name);
-  auto pos_package_name = topic_name.find(service);
-  if (pos_type_name == std::string::npos || pos_package_name == std::string::npos) {
+  auto service_lowercase = type_name.substr(0, pos_msg);
+  std::string topic_name_lowercase = topic_name;
+  transform(topic_name_lowercase.begin(), topic_name_lowercase.end(), topic_name_lowercase.begin(), ::tolower);
+  auto pos_package_name = topic_name_lowercase.find(service_lowercase);
```

Comment:
I find this change a little hard to read. I would actually propose to change the API of this call from `const std::string & topic_name, const std::string & type_name` to take the string by values. This would allow us to simply call

```
topic_name = std::transform(topic_name.begin(), topic_name.end(), topic_name.begin(), ::tolower);
type_name = std::transform(type_name.begin(), type_name.end(), type_name.begin(), ::tolower);
```

and the logic on how to parse/extract the tuple stays the same.

Reply:
This way we would lose the information whether the `service` name had capital letters or not. I'll try to make it more selfexplanatory.